web/src/views/repo/RepoBuild.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+  <Panel>
+    <InputField :label="$t('manual.select_branch')">
+      <SelectField
+        v-model="this.payload.branch"
+        :options="this.branches"
+        :disabled="loading"
+        required
+        class="bg-transparent text-color"
+      />
+    </InputField>
+    <div>
+      <InputField :label="$t('manual.variable_key')">
+        <TextField
+          v-model="this.tmpVar.key"
+          :placeholder="$t('manual.var_key')"
+          required
+          :disabled="loading"
+        />
+      </InputField>
+      <InputField :label="$t('manual.variable_value')">
+        <TextField
+          v-model="this.tmpVar.value"
+          :placeholder="$t('manual.var_value')"
+          required
+          :disabled="loading"
+        />
+      </InputField>
+      <Button
+        :is-loading="loading"
+        type="submit"
+        :text="$t('manual.add_variable')"
+        @click="addVar()"
+      />
+    </div>
+    <br>
+    <div class="text-color">
+      <div v-for="(v, k) in this.payload.variables" :key="k">
+      <pre><Button
+        type="submit"
+        text="X"
+        @click="deleteVar(k)"
+        style="display: inline-block;"
+      />&nbsp;<span class="font-bold">{{ k }}</span>&#9;{{ v }}</pre>
+      </div>
+    </div>
+    <br>
+    <Button
+      :is-loading="loading"
+      type="submit"
+      :text="$t('manual.launch_build')"
+      @click="runManual()"
+    />
+  </Panel>
+</template>
+
+<script lang="ts">
+import {defineComponent, inject, onMounted, Ref} from 'vue';
+
+import IconButton from '~/components/atomic/IconButton.vue';
+import useApiClient from '~/compositions/useApiClient';
+
+import InputField from '~/components/form/InputField.vue';
+import TextField from '~/components/form/TextField.vue';
+import SelectField from "../../components/form/SelectField.vue";
+import Panel from "../../components/layout/Panel.vue";
+
+const apiClient = useApiClient();
+
+export default defineComponent({
+  name: 'RepoBuild',
+
+  components: {
+    Panel,
+    SelectField,
+    IconButton,
+    InputField,
+    TextField
+  },
+
+  setup() {
+
+  },
+
+  mounted() {
```

Comment:
Could you migrate this to composition api with setup? If you need help just ask, I would be happy to support you. 

For starting a manual pipeline I would suggest having a button like this at the top:

![image](https://user-images.githubusercontent.com/6918444/188205406-0bf10e41-1e79-41ac-a800-dae5a8eeaf27.png)

Which opens a popup like this:

![Screenshot from 2022-09-02 19-14-28](https://user-images.githubusercontent.com/6918444/188205742-b2870186-911b-4a73-84c3-6fc29ade9744.png)

Reply:
i did not find any popup to gather inspiration from inside woodpecker codebase, do you have any suggestion?

I can still implement that from scratch, but i would like to avoid reinventing the wheel

Reply:
@anbraten migrated all to composition api

Reply:
I don't think we have a popup yet, but it should be quite simple to implement base on something like this:

```vue
<template>
  <!-- overlay -->
  <div
    v-if="open"
    class="fixed bg-gray-900 opacity-80 left-0 top-0 right-0 bottom-0 z-500 print:hidden"
    @click="$emit('close')"
  />
  <!-- overlay end -->
  <transition :name="`slide-${position}`" class="print:hidden fixed flex left-0 top-0 right-0 bottom-0">
    <div
      v-if="open"
      class="m-auto flex flex-col bg-gray-200 shadow-all z-1000 max-w-64"
    >
      <slot />
    </div>
  </transition>
</template>

<script lang="ts">
import { defineComponent, PropType, ref, toRef, watch } from 'vue';

export default defineComponent({
  name: 'Popup',

  props: {
    open: {
      type: Boolean,
    },
  },

  emits: {
    close: () => true,
  },
});
</script>

<style scoped>
.slide-right-enter-active,
.slide-right-leave-active {
  transition: all 0.3s ease;
}
.slide-right-enter-from,
.slide-right-leave-to {
  transform: translate(100%, 0);
}

.slide-bottom-enter-active,
.slide-bottom-leave-active {
  transition: all 0.3s ease;
}
.slide-bottom-enter-from,
.slide-bottom-leave-to {
  transform: translate(0, 100%);
}
</style>
```

Reply:
for the moment i have added the button to the right that loads the page in the router-view as before:
![image](https://user-images.githubusercontent.com/1896292/188215211-084cdfe1-dd64-4e37-8391-8c7bb22c9d3b.png)

a popup is not so usefull as after trigger the page will be redirected to the created pipeline